### PR TITLE
Fixed vehicle explosion and missile impact sounds in Tiberian Dawn

### PR DESCRIPTION
**Pull request by VonNah**
I crossreferenced all the original sounds and noticed some oddities.

I fixed most of them by editing explosions, missiles and vehicles.yaml and thoroughly tested them by comparing the mod with CnC Net's Tiberian Dawn or CnC Remastered Collection. 

OpenRA Tiberian Dawn has more sound variety now which not only polishes the mod and its single player campaign, but also makes multiplayer more enjoyable.

### CODE REVIEW

File: mods/cnc/rules/vehicles.yaml [hunk 1]
```diff
@@ -512,6 +518,9 @@ HTNK:
 		Delay: 10
 		StartIfBelow: 50
 		DamageCooldown: 200
+	Explodes:
+		Weapon: ArtilleryShell
```
Comment (pchote) [inline mods/cnc/rules/vehicles.yaml]: Shouldn't we be adding a new `UnitExplodeArtillery` for this?

Comment (VonNah) [reply mods/cnc/rules/vehicles.yaml]: I tested it and it works, but keep in mind that Mammoth Tanks must also have the same effects. So is it okay to add `UnitExplodeArtillery` to them too? I already added a new `UnitExplodeBig` with different effects.

Comment (pchote) [reply mods/cnc/rules/vehicles.yaml]: Ah, i see. We should rename this to something more clearly related to the animation/sound then.

Comment (VonNah) [reply mods/cnc/rules/vehicles.yaml]: If I'm not mistaken only Nod's Artillery, GDI' MLRS (which are called MSAM in the vehicles.yaml btw) and Mammoth Tanks share the same explosion/sound effects.

**Comment by VonNah** [comment]
I successfully added the Explodes@Empty definition (Thank you pchote). This PR closes #18975

**Comment by Smittytron** [comment]
Can you please squash the two new commits under their respective commits? The [contributing](https://github.com/OpenRA/OpenRA/wiki/Contributing) guide has an explainer on how to do this.

**Comment by VonNah** [comment]
Something went wrong. At this point I don't know if I have to make the same pull request again or if I can add those files back in here. I am stll learning how to use github. Sorry for the inconvenience.

**Comment by phrohdoh** [comment]
> Something went wrong. At this point I don't know if I have to make the same pull request again or if I can add those files back in here. I am stll learning how to use github. Sorry for the inconvenience.

I personally would:
- reset this branch to current upstream `bleed` (which will throw away any and all changes locally/on this branch/etc, `git reset --hard upstream/bleed` assuming `upstream` is the name of your remote pointing to openra proper)
- make & commit your changes again
- force-push your local branch over your remote branch, which will update this PR (`git push origin +myfeature`)

It'll probably be easier to help with this via Discord (or IRC, something realtime), so feel free to ping me (or just ask for git help in general) there if you want.

**Comment by VonNah** [comment]
I renamed `ArtilleryShell` to `UnitExplodeMech` and added the definition to vehicles.yaml. Why Mech?. Westwood Studios described `xplosml2.aud` as a _muffled mechanical explosion_ .

**Review by phrohdoh — DISMISSED** [review]
LGTM

File: mods/cnc/weapons/explosions.yaml [hunk 2]
```diff
@@ -56,7 +56,7 @@ UnitExplode:
 			Heavy: 24
 	Warhead@2Eff: CreateEffect
 		ImpactSounds: xplobig6.aud
-
+
```
Comment (pchote) [inline mods/cnc/weapons/explosions.yaml]: Stray whitespace addition

**Comment by VonNah** [comment]
Updated

File: mods/cnc/weapons/missiles.yaml [hunk 3]
```diff
@@ -237,6 +240,7 @@ TowerMissile:
 		DamageTypes: Prone50Percent, TriggerProne, SmallExplosionDeath
 	Warhead@3Eff: CreateEffect
 		Explosions: med_frag
+		ImpactSounds: xplobig4.aud
```
Comment (pchote) [inline mods/cnc/weapons/missiles.yaml]: I don't think this works for the AGT as it is currently balanced. The explosion from the first missile hasn't even finished before the next one starts playing!

Comment (VonNah) [reply mods/cnc/weapons/missiles.yaml]: You are right. Its fire rate is too high. Should I remove `xplobig4` and rebase again?

Comment (pchote) [reply mods/cnc/weapons/missiles.yaml]: That would probably best, but it would be good to get some opinions from some of our active TD players IMO.

**Comment by VonNah** [comment]
Updated. Changing the AGT firerate (for the original explosion sound) will have large balance consequences. It needs a lot of testing.

**Comment by reaperrr** [comment]
In my opinion, changing the sound only makes sense if we also change the explosion effect to the one it was tied to in the original.
In this case that would be `big_frag`, which was used for those missiles in the original and better matches the sound in terms of length, too.

**Review by pchote — APPROVED** [review]
I wouldn't be surprised if some of the changes (across all the sound PRs, not just this one) draw complaints, but IMO these are worth a shot and we can always undo specific changes based on feedback on the next playtest.

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/c9055c87685c7fd9d6e20494ac24f54c1f38b9f0

90ddbbc934

**Pull request merged**